feat: Implemented /chat-summary help command

dhairyashiil opened the pull request:
This PR introduces a new `/chat-summary help` command designed to assist users in understanding how to use the Chat Summarizer app. The helper message is displayed in a RocketChat-style format and includes the following sections:

- A welcoming message to greet users and introduce the app.
- Instructions on how to set up the LLM (Large Language Model) endpoint, including options for internal models, external models like OpenAI or Gemini, and self-hosted models.
- A list of available commands and what each one does.
- Details about available add-ons, such as file summarization, and how to enable them in the app settings.

Additionally, the helper message aligns with the README, expanding the "How to Use" section to include detailed instructions on using available commands and add-ons.
Updated the README as well.

Closes #21

Video:

https://github.com/user-attachments/assets/2930995a-567d-426c-bc4d-29992ee75dff

dhairyashiil committed:
Implemented /chat-summary help command

dhairyashiil committed:
Implemented /chat-summary help command

dhairyashiil commented:
Hello @JeffreytheCoder , Please review the PR.

One thing I want to ask is about the welcome message. Do we really need to send the welcome message every time, or should we keep a record of whether the user has already used the app in IPersistence storage? However, I think this could add unnecessary calculations. Personally, I believe we should drop the welcome message part. What do you think?

JeffreytheCoder commented on app/constants/dialogue.ts:
```diff
@@ -0,0 +1,15 @@
+
+Cheers,
+The Rocket.Chat Team ✨
+`;
```
Love the style! However, the user may use the app in servers other than the RC open server. We can add one simple welcome message combined with the help answer

JeffreytheCoder commented on app/constants/prompts.ts:
```diff
@@ -215,3 +215,91 @@ Only give the output using the format above and nothing else.
+
+export function createUserHelpPrompt(userQuestion: string): string {
+	return HELP_USER_PROMPT.replace('{userQuestion}', userQuestion);
+}
```
Really like the way of using AI to answer user's questions! 

However, in the prompt, it seems LLM doesn't enough context, e.g. about what RC app is, what this app does, what app settings are. In your demo, we see LLM providing answer not directly related to "how to configure to app".

We do want to have a `/help` command that shows a fix answer so the user gets a direct understanding, and `/help <question>` gives answer by AI (with better prompt engineered)

JeffreytheCoder commented:
> Hello @JeffreytheCoder , Please review the PR.
> 
> One thing I want to ask is about the welcome message. Do we really need to send the welcome message every time, or should we keep a record of whether the user has already used the app in IPersistence storage? However, I think this could add unnecessary calculations. Personally, I believe we should drop the welcome message part. What do you think?

@dhairyashiil Thanks for implementing this so quickly! We can have one brief welcome message combined with the (non-AI) help answer so we don't send two messages every time

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

dhairyashiil commented:
Hello @JeffreytheCoder , 

1. Updated the welcome message.
2. Introduced two types of help commands:
    `/chat-summary help`
    `/chat-summary help <question>`
3. Improved the prompt for AI responses.

`/chat-summary help` -> Displays the welcome message and Frequently Asked Questions.
`/chat-summary help <question>` -> Provides an answer from the AI.


`/chat-summary help`



https://github.com/user-attachments/assets/21d47acc-a569-4da8-83cc-86908ba8afd0





`/chat-summary help <question>`


https://github.com/user-attachments/assets/5efcb2e2-0809-479d-bc76-b7bb4d9473c3

dhairyashiil committed:
Implemented 2 help commands with fixed and AI reply

JeffreytheCoder commented:
@dhairyashiil LGTM! Great work 🚀

JeffreytheCoder commented:
@dhairyashiil Feel free to solve the conflicts and merge this PR

dhairyashiil committed:
Resolved merge conflicts

dhairyashiil commented:
Hello @JeffreytheCoder , Resolved the merge conflicts. please check

JeffreytheCoder commented:
@dhairyashiil Looking good! Merging this PR

dhairyashiil commented:
> @dhairyashiil LGTM! Great work 🚀

Getting this comment from you means a lot to me. Thank you 🫠
